### PR TITLE
Feature/bai 1343 add announcement banner

### PR DESCRIPTION
**Pull request by ARADDCC002**

### CODE REVIEW

**Review by GB27247 — CHANGES_REQUESTED** [review]

File: frontend/src/Annoucement.tsx [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+            <Button variant='text' size='small' onClick={() => setShowFullText(!showFullText)}>
+              {showFullText ? 'Show less' : 'Show more'}
+            </Button>
+          </span>
+        </Typography>
+      </Stack>
+    ) : (
+      message
+    )
+  }
```
[the remark this answers is not in the document]
Comment (ARADDCC002) [reply frontend/src/Annoucement.tsx:31]: This is a mistake and should be memozied, updated!

File: frontend/src/Wrapper.tsx [hunk 2]
```diff
@@ -105,6 +121,16 @@ export default function Wrapper({ children }: WrapperProps): ReactElement {
               </>
             )}
           </Box>
+          {isUiConfigLoading && <Loading />}
+          {uiConfig && (
+            <Box sx={{ mb: 2 }}>
+              {annoucementBannerOpen ? (
+                <Announcement message={uiConfig.announcement.text} onClose={handleAnnouncementOnClose} />
```
[the remark this answers is not in the document]
Comment (ARADDCC002) [reply frontend/src/Wrapper.tsx:128]: We discussed this and chose to go at the bottom to keep it consistent with other applications with similar messages. If enough users prefer it moved we can look to doing that.

**Review by GB27247 — DISMISSED** [review]

File: frontend/src/Banner.tsx [hunk 3]
```diff
@@ -24,7 +25,11 @@ function Banner() {
     return null
   }
 
-  return <Card sx={style}>{uiConfig.banner.text}</Card>
+  return (
+    <Card sx={style}>
+      <Stack>
```
Comment (ARADDCC012) [inline frontend/src/Banner.tsx:30]: This stack isn't needed as it only has one child

File: frontend/src/Wrapper.tsx [hunk 4]
```diff
@@ -17,6 +19,8 @@ export type WrapperProps = {
   children?: ReactNode
 }
 
+const DISMISSED_COOKIE_NAME = 'dismissed-timestamp'
```
Comment (ARADDCC012) [inline frontend/src/Wrapper.tsx:22]: Can we move this into `frontend/utils/constants.ts`?

File: frontend/src/Wrapper.tsx [hunk 5]
```diff
@@ -44,6 +51,16 @@ export default function Wrapper({ children }: WrapperProps): ReactElement {
     }
   }, [isUiConfigLoading, uiConfig, isDocsPage])
 
+  useEffect(() => {
+    if (uiConfig) {
+      setAnnouncementBannerOpen(
+        uiConfig &&
```
Comment (ARADDCC012) [inline frontend/src/Wrapper.tsx:57]: We've already checked `uiConfig` is truthy above so this check isn't needed

File: frontend/src/Wrapper.tsx [hunk 6]
```diff
@@ -2,8 +2,10 @@ import Box from '@mui/material/Box'
 import { useTheme } from '@mui/material/styles'
 import Toolbar from '@mui/material/Toolbar'
 import { useGetUiConfig } from 'actions/uiConfig'
+import Cookies from 'js-cookie'
```
Comment (ARADDCC012) [inline frontend/src/Wrapper.tsx:5]: I know the library uses `Cookies` in their examples, but can we use `cookies` as it then follows our naming conventions?

File: frontend/src/Announcement.tsx [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+          </Button>
+        </Typography>
+      </Stack>
+    ) : (
+      message
+    )
+  }, [message, showFullText])
+
+  return (
+    <>
```
Comment (ARADDCC012) [inline frontend/src/Announcement.tsx:32]: This fragment isn't needed

File: frontend/src/Announcement.tsx [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+          bottom: 0,
+          mb: 4,
+          borderStyle: 'solid',
+          borderWidth: 2,
+          borderColor: theme.palette.primary.main,
+          borderRadius: 1,
+          left: 0,
+          right: 0,
+          maxWidth: 'md',
+          ml: 'auto',
```
Comment (ARADDCC012) [inline frontend/src/Announcement.tsx:47]: Could combine `ml` and `mr` by using `mx: 'auto'`

File: frontend/src/Announcement.tsx [hunk 9]
```diff
@@ -0,0 +1,80 @@
...
+export default function Announcement({ message, onClose }: AnnoucementProps) {
+  const theme = useTheme()
+
+  const [showFullText, setShowFullText] = useState(false)
+
+  const announcementText = useMemo(() => {
+    return message.length > 100 ? (
+      <Stack>
+        <Typography>
+          {!showFullText ? `${message.slice(0, 100)}...` : message}
```
Comment (ARADDCC012) [inline frontend/src/Announcement.tsx:20]: Can we reverse the logic here? It's a little confusing to have a ternary written this way round. Consider:
```
{showFullText ? message : `${message.slice(0, 100)}...`}
```

File: frontend/src/Wrapper.tsx [hunk 10]
```diff
@@ -105,6 +127,14 @@ export default function Wrapper({ children }: WrapperProps): ReactElement {
               </>
             )}
           </Box>
+          {isUiConfigLoading && <Loading />}
+          {uiConfig && (
+            <Box sx={{ mb: 2 }}>
```
Comment (ARADDCC012) [inline frontend/src/Wrapper.tsx:132]: Is this box necessary? I don't think we want the banner to take up any space from the existing content, we want it to sit on top of it.

Also consider combining the two conditional blocks into:
```
{uiConfig && annoucementBannerOpen && ( ... )}
```

**Review by ARADDCC012 — CHANGES_REQUESTED** [review]

**Review by ARADDCC012 — APPROVED** [review]

**Pull request merged**